file_store.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/oras-project/oras-credentials-go/internal/ioutils"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// FileStore implements a credentials store using the docker configuration file
+// to keep the credentials in plain-text.
+type FileStore struct {
+	// DisableSave disable saving credentials in plain text.
+	// If DisableSave is set to true, Put() will return ErrPlainTextSaveDisabled.
+	DisableSave   bool
```

Comment:
`readonly` might not be appropriate. This flag does not make the store "read only", as the users can still delete credentials when the flag is set.

Reply:
Maybe disable delete as well and ErrPlainTextSaveDisabled could be permission denied.

Reply:
The idea of this feature is to avoid saving credentials in plain text for security consideration,  while deleting plain-text credentials doesn't seem to harm security? @shizhMSFT What do you think?

Reply:
Deleting sounds good to me.